src/components/BuySellForm/BuySellForm.tsx

```diff
@@ -261,6 +263,17 @@ const BuySellForm = ({ advertId, isModalOpen, onRequestClose }: TBuySellFormProp
         if (!isModalOpenFor('RateFluctuationModal')) finalEffectiveRateRef.current = effectiveRate;
     }, [effectiveRate, hasRateChanged, isModalOpenFor]);
 
+    useEffect(() => {
+        if (
+            balanceData.balance !== undefined &&
+            minOrderAmountLimit !== undefined &&
+            isBuy &&
+            (balanceData.balance === 0 || balanceData.balance < minOrderAmountLimit)
+        )
+            setShowLowBalanceError(true);
+        else setShowLowBalanceError(false);
```

Comment:
```suggestion
        const shouldShowError = (
            balanceData.balance !== undefined &&
            minOrderAmountLimit !== undefined &&
            isBuy &&
            (balanceData.balance === 0 || balanceData.balance < minOrderAmountLimit)
        );
        setShowLowBalanceError(shouldShowError);
```

will this work?

Reply:
yup sure, this can make it more readable